contracts/CollateralFundV1.sol

```diff
@@ -2,131 +2,155 @@
-    uint256 public totalCollateral;
+contract CollateralFundV1 {
+    // Fund parameters.
+    mapping(address => uint256) public addressToCollateralAmount;
+    mapping(address => bool) public addressToFunderStatus; // It returns false if the user is not in the funders list.
     address[] public funders;
 
-    ERC20Token public usdcKovan; 
+    // USDC token parameters.
+    address public usdcAddress;
+    IERC20 public usdcContract;
 
     // Access-control parameters.
-    address hubAddress;
+    address public owner;
+    address public hubAddress;
 
     // Modifiers
-    modifier onlyHub() {
-        require(msg.sender == hubAddress, "Only the hub can call this method");
+    modifier onlyOwner() {
+        require (msg.sender == owner, "Only the owner can call this method");
         _;
     }
 
     // Events
-    event Deposit(address indexed _addr, uint256 _amount, uint256 _balance);
-    event Withdraw(address indexed _addr, uint256 _amount, uint256 _balance);
+    event Deposit(address indexed addr, uint256 amount, uint256 updatedUserCollateralAmount);
+    event Withdraw(address indexed addr, uint256 amount, uint256 updatedUserCollateralAmount);
+    event CollateralAmountUpdated(address indexed addr, uint256 previousAmount, uint256 newAmount);
+    event NewFunder(address addr);
+    event USDCTokenAddressUpdated(address oldAddress, address newAddress);
+    event HubAddressUpdated(address oldAddress, address newAddress);
 
     /**
      * @notice Initialise the contract.
+     * @param _usdcAddress the address of the USDC token contract.
      * @param _hubAddress the address of the hub.
      */
-    constructor(address _hubAddress) {
+    //slither-disable-next-line naming-convention
+    constructor(address _usdcAddress, address _hubAddress) {
+        owner = msg.sender;
+        usdcAddress = _usdcAddress;
+        usdcContract = IERC20(_usdcAddress);
         hubAddress = _hubAddress;
     }
 
+    /**************************************** Fund / Withdraw ****************************************/
+
     /**
-     * @notice Send collateral to the fund.
+     * @notice Deposit collateral to the fund.
+     * @param _amount the amount of USDC token send to the fund.
      */
-    function fund() external payable {
-        // TODO: change to USDC instead of Ether
-        collateralFundedByAddress[msg.sender] += msg.value;
-        totalCollateral += msg.value;
-        funders.push(msg.sender);
-        emit Deposit(
-            msg.sender,
-            msg.value,
-            collateralFundedByAddress[msg.sender]
-        );
+    //slither-disable-next-line naming-conventions
+    function deposit(uint256 _amount) external {
+        require(_amount > 0, "Amount should be greator than zero");
+
+        // Send the USDC from the user's wallet to the fund and update the user collateral amount.
+        usdcContract.transferFrom(msg.sender, address(this), _amount);
```

Comment:
maybe stupid question, but how is our CollateralFundV1 address approved?

Reply:
We'll do this in the front-end app, check `test/collateral-fund.js`.